src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/external/SentenceCalculationType.kt

```diff
@@ -1,172 +1,197 @@
-  SEC94(isSupported = false, isIndeterminate = true),
-  SEC93_03(isSupported = false, isIndeterminate = true),
-  ALP_CODE18(isSupported = false, isIndeterminate = true),
-  DPP(isSupported = false, isIndeterminate = true),
-  SEC272(isSupported = false, isIndeterminate = true),
-  SEC275(isSupported = false, isIndeterminate = true),
-  ALP_CODE21(isSupported = false, isIndeterminate = true),
-  LR_DPP(isSupported = false, isIndeterminate = true),
-  LR_ALP_CDE21(isSupported = false, isIndeterminate = true),
-  LR_ALP_LASPO(isSupported = false, isIndeterminate = true),
-  ZMD(isSupported = false, isIndeterminate = true),
-  SEC93(isSupported = false, isIndeterminate = true),
-  TWENTY(isSupported = false, isIndeterminate = true, primaryName = "20"),
-  NP(isSupported = false, isIndeterminate = false),
-  LR_EPP(isSupported = false, isIndeterminate = false),
-  CR(isSupported = false, isIndeterminate = false),
-  BOTUS(isSupported = true, isIndeterminate = false, sentenceClazz = BotusSentence::class.java),
-  AR(isSupported = false, isIndeterminate = false),
-  EPP(isSupported = false, isIndeterminate = false),
-  CUR_ORA(isSupported = false, isIndeterminate = false),
-  A_FINE(isSupported = false, isIndeterminate = false, primaryName = "A/FINE"),
-  CUR(isSupported = false, isIndeterminate = false),
-  CIVIL(isSupported = false, isIndeterminate = false),
-  EXT(isSupported = false, isIndeterminate = false),
-  LR_ES(isSupported = false, isIndeterminate = false),
-  YRO(isSupported = false, isIndeterminate = false),
-  SEC91(isSupported = false, isIndeterminate = false),
-  VOO(isSupported = false, isIndeterminate = false),
-  TISCS(isSupported = false, isIndeterminate = false),
-  STS21(isSupported = false, isIndeterminate = false),
-  STS18(isSupported = false, isIndeterminate = false),
-  UNIDENTIFIED(isSupported = false),
+  HDR_ORA(sentenceType = UNIMPLEMENTED_SENTENCE_TYPE, recallType = STANDARD_RECALL_255),
+  HDR(sentenceType = UNIMPLEMENTED_SENTENCE_TYPE, recallType = STANDARD_RECALL_255),
+  CUR(sentenceType = UNIMPLEMENTED_SENTENCE_TYPE, recallType = STANDARD_RECALL_255),
+  CUR_ORA(sentenceType = UNIMPLEMENTED_SENTENCE_TYPE, recallType = STANDARD_RECALL_255),
+  //endregion
+
+  //region Unidentified
+  UNIDENTIFIED(sentenceType = UNIMPLEMENTED_SENTENCE_TYPE),
+  //endregion
   ;
 
   companion object {
+
     fun from(sentenceCalculationType: String): SentenceCalculationType =
       entries.firstOrNull { it.primaryName == sentenceCalculationType }
         ?: entries.firstOrNull { it.name == sentenceCalculationType }
         ?: UNIDENTIFIED
 
-    fun isSupported(sentenceCalculationType: String): Boolean =
-      try {
-        from(sentenceCalculationType).isSupported
-      } catch (error: IllegalArgumentException) {
-        false
-      }
+    fun isCalculable(sentenceCalculationType: String): Boolean =
+      !(isIndeterminate(sentenceCalculationType) || isUnimplementedSentence(sentenceCalculationType))
```

Comment:
I think I prefer `from(sentenceCalculationType).sentenceType?.sentenceClazz != null` but okay if you think this is better.